File: 2021/day01.go
package main

import (
	"fmt"
	"github.com/jjbeto/adventofcode/helper"
)

func part1(numbers []int64) int64 {
	var prev, increases int64 = -1, 0
	for i, next := range numbers {
		if prev < next && i > 0 {
			increases++
		}
		prev = next
	}
	return increases
}

func part2(numbers []int64) int64 {
	var rolling []int64
	for i := 0; i <= len(numbers)-2; i++ {
		rolling = append(rolling, helper.Sum(numbers[i:i+3]))
	}
	return part1(rolling)
}

func main() {
	numbers := helper.ReadNumbers("2021", "day01.data", 10)
	fmt.Println(part1(numbers))
	fmt.Println(part2(numbers))
}

File: 2021/day01.py
from aocd import get_data

raw = get_data(day=1, year=2021)

numbers = list([int(n) for n in raw.splitlines()])


def increases(seq):
    return sum(1 for n in range(len(seq) - 1) if seq[n + 1] > seq[n])


print(f'Answer 1: {increases(numbers)}')

rolling = [sum(numbers[i:i + 3]) for i in range(len(numbers) - (3 - 1))]
print(f'Answer 2: {increases(rolling)}')
